### DISCUSSION TITLE
How to get template for the different iTwin Viewer for each Open technology design tool/discipline?

### DISCUSSION THREAD

**Discussion by JiriJa** [iTwin/itwinjs-core#2676] [Q&A] [answered]
Hello everyone! What do You think, does Bentley provide template for each discipline specific "default iTwin viewer"? This could give me sense, that export via iTwinSynchronizer directly from each native Open* technology product will produce different iTwin viewer depends on default discipline Open environment … **What do You think, is there anywhere right repository with plenty of iTwin viewer templates with respect on which discipline model was born from?**  :-) Because Time _is money_ / or effort :-) and why to begin from absolute scratch core?

Note 1: I´m OpenRoads Highway designer. My goal is: let this will be my starting point and I will try to modify them  and add many “roads” view attributes and Widgets ... modules/extensions that will be focused on specifics  to the local authorities and departments of transportation.

Note 2: I´m new here with 2 months of experience as an Bentley Accredited Developer iTwin Platform

![01_OpenRoads_iTwin_Ribon](https://user-images.githubusercontent.com/34089657/141069682-cce9962d-5bf1-4212-b3c6-77ce641426ff.jpg)

**Default iTwinViewer for OpenRoads:**
![02_Deafult_iTwinViewer_depends_on_ORD](https://user-images.githubusercontent.com/34089657/141069701-181e4bf8-9845-4a7a-8fc4-3b990e410d51.jpg)
.

**Comment by kckst8** [comment, marked as answer]
Hi @JiriJa. We have a goal of providing more use case specific templates with different sets of default tooling and widgets and/or a more interactive template system. This is most likely a Q1 2022 reality, however.

In the meantime, the new 3.0 extension APIs will make it easier to provide reusable extensions and begin to build a library to help avoid duplication.

**Reply by JiriJa** [reply]
Hi @kckst8. Thanks ...  the road plan with date Q1 2022 sounds good ... if it is realistic.
-Jiri

**Reply by JiriJa** [reply]
Hi  @kckst8 , what do You think ... are now better conditions for looking for "case specific iTwin viewer templates with different sets of default tooling and widgets"? ... suited for civil infrastructure? ... for discipline as "_OpenRoads_ **Highway** designer". Thanks Jiri.

**Reply by aruniverse** [reply]
We've started looking into this, looks more like a Q2 reality now. Just curious, what specific features are you looking for?

**Reply by JiriJa** [reply]
... this "specific features" are looking by DoT and Municipalities **end** users :-) 
top list of them: dynamic long profile view, dynamic perpendicular cross section view, cross section by 2 points.
